web/controllers/auth.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Healthlocker.Auth do
+  import Plug.Conn
+  import Comeonin.Bcrypt, only: [checkpw: 2, dummy_checkpw: 0]
+
+  def init(opts) do
+    Keyword.fetch!(opts, :repo)
+  end
+
+  def call(conn, repo) do
+    user_id = get_session(conn, :user_id)
+    user = user_id && repo.get(Healthlocker.User, user_id)
```

Comment:
The `&&` tends to be a bit of a code smell. If the `user_id` is nil shouldn't we just error? Maybe we can use pattern matching on the `get_session` line to catch it?